pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -25,319 +25,356 @@ import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/IERC20Permit.sol
+     * @dev Returns the timestamp up to which a distribution has been distributing tokens
+     * @param distributionId ID of the distribution being queried
      */
-    function getDistributionId(
-        IERC20 stakingToken,
-        IERC20 rewardsToken,
-        address rewarder
-    ) public pure returns (bytes32) {
-        return keccak256(abi.encodePacked(stakingToken, rewardsToken, rewarder));
+    function lastTimePaymentApplicable(bytes32 distributionId) public view override returns (uint256) {
+        return _lastTimePaymentApplicable(_getDistribution(distributionId));
     }
 
     /**
-     * @dev Tells the information of a distribution
+     * @dev Returns if a user is subscribed to a distribution or not
      * @param distributionId ID of the distribution being queried
+     * @param user The address of the user being queried
      */
-    function getDistribution(bytes32 distributionId) external view returns (Distribution memory) {
-        return _getDistribution(distributionId);
+    function isSubscribed(bytes32 distributionId, address user) external view override returns (bool) {
+        IERC20 stakingToken = _getDistribution(distributionId).stakingToken;
+        return _userStakings[stakingToken][user].subscribedDistributions.contains(distributionId);
     }
 
     /**
-     * @dev Tells the information of a distribution for a user
+     * @dev Returns the information of a distribution for a user
      * @param distributionId ID of the distribution being queried
      * @param user Address of the user being queried
      */
-    function getUserDistribution(bytes32 distributionId, address user) external view returns (UserDistribution memory) {
+    function getUserDistribution(bytes32 distributionId, address user)
+        external
+        view
+        override
+        returns (UserDistribution memory)
+    {
         IERC20 stakingToken = _getDistribution(distributionId).stakingToken;
         return _userStakings[stakingToken][user].distributions[distributionId];
     }
 
     /**
-     * @dev Tells if a user is subscribed to a distribution or not
+     * @dev Returns the unaccounted earned payment for a user until now for a particular distribution
      * @param distributionId ID of the distribution being queried
-     * @param user The address of the user being queried
+     * @param user Address of the user being queried
      */
-    function isSubscribed(bytes32 distributionId, address user) external view returns (bool) {
+    function unaccountedUnclaimedTokens(bytes32 distributionId, address user) external view override returns (uint256) {
         IERC20 stakingToken = _getDistribution(distributionId).stakingToken;
-        return _userStakings[stakingToken][user].allowedDistributions.contains(distributionId);
+        UserStaking storage userStaking = _userStakings[stakingToken][user];
+        return _unaccountedUnclaimedTokens(userStaking, distributionId);
```

Comment:
This roughly lines up with my thoughts. I think they're accessible purely because they were in the original contract but for some of these I don't see a reason to expose them.

Reply:
Agreed, created https://github.com/balancer-labs/balancer-v2-monorepo/issues/1016